src/pages/myCohort/index.js

```diff
@@ -1,27 +1,41 @@
-import { useEffect, useState } from "react"
+import { useCallback, useContext, useEffect, useState } from "react"
 import MyExercises from "../../components/myCohortExercises"
 import MyTeachers from "../../components/myCohortTeachers"
 import MyCohortDetails from "../../components/myCohortDetails"
-import { getUsers, getSelf } from "../../service/apiClient"
+import { getStudentsByCohortId, getTeachers } from "../../service/apiClient"
+import { AuthContext } from "../../context/auth"
 
 const MyCohort = () => {
-  const [users, setUsers] = useState([])
+  const [students, setStudents] = useState([])
+  const [teachers, setTeachers] = useState([])
 
-  useEffect(() => {
-    getUsers().then((users) => {
-			getSelf()
-				.then((self) => setUsers(users.filter((user) => user.cohort_id === self.cohort_id &&	user.role === "STUDENT")))
-		})
+  const loggedInStudent = useContext(AuthContext).loggedInStudent
+
+  const getMyClassmates = useCallback((cohortId) => {
+    getStudentsByCohortId(cohortId).then(setStudents)
   }, [])
 
+  const getAllTeachers = useCallback(() => getTeachers().then(setTeachers), [])
+
+  useEffect(() => {
+    getAllTeachers()
+
+    if (loggedInStudent) {
+      getMyClassmates(loggedInStudent.cohortId)
```

Comment:
@yee0802 Could it happen that `cohortId` is undefined? In that case, you might end up with a faulty API call

Reply:
Yeah, I understand now. 
Just seen Chloes PR had a similar issue so it *should* be gtg now